### ISSUE TITLE
ERROR Daemon Failed to run 'daemon': Traceback (most recent call last)... under SLES 12 SP3

### ISSUE THREAD

**Issue by iszabi** [Azure/WALinuxAgent#1238]
Hi,

After I have installed WALinuxAgent on SLES 12 SP3 (up-to-date OS)
There are many errors in /var/log/waagent.log 

```
--
2018/07/04 13:02:43.566209 INFO Daemon Azure Linux Agent Version:2.2.27
2018/07/04 13:02:43.596381 INFO Daemon OS: suse 12
2018/07/04 13:02:43.619839 INFO Daemon Python: 2.7.13
2018/07/04 13:02:43.643497 ERROR Daemon Failed to run 'daemon': Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/agent.py", line 168, in main
    agent.daemon()
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/agent.py", line 99, in daemon
    daemon_handler.run(child_args=child_args)
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/daemon/main.py", line 69, in run
    CGroups.setup()
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/common/cgroups.py", line 536, in setup
    log_event=False)
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/common/event.py", line 402, in add_event
    is_internal=is_internal, log_event=log_event)
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/common/event.py", line 262, in add_event
    self._add_event(duration, evt_type, is_internal, is_success, message, name, op, version, eventId=1)
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/common/event.py", line 279, in _add_event
    self.save_event(json.dumps(data))
  File "/usr/lib/python2.7/site-packages/azurelinuxagent/common/event.py", line 204, in save_event
    os.mkdir(self.event_dir)
OSError: [Errno 2] No such file or directory: '/var/lib/waagent/events'
---

```
This errors have been been showed in Azure Serial Console (Preview), too.

What do you suggest, how can I solve this issue?

Thank you in advance.

Best Regards,
Szabolcs Illes

**Comment by boumenot** [comment]
We have to figure out why this happening, but I can give you a mitigation in the interim.

```sh
mkdir /var/lib/waagent/events
```

**Comment by grafi-tt** [comment]
This would be fixed by #1246.

**Comment by hglkrijger** [comment]
Fix merged
